admin/tests/test_testing.py

```diff
@@ -0,0 +1,436 @@
+
+        # make file with some data
+        with tempfile.TemporaryDirectory() as tmpdir:
+            fname = os.path.join(tmpdir, "data.txt")
+            with open(fname, "w") as f:
+                f.write("1 10\n 20\t 30\n42\n")
+
+            # random mode
+            gen = config._ValueIntFromFile(fname)
+            values = set(gen() for i in range(100))
+
+            for val in values:
+                self.assertIn(val, {1, 10, 20, 30, 42})
+
+            # sequential mode
+            gen = config._ValueIntFromFile(fname, "sequential")
+            values = [gen() for i in range(10)]
+            self.assertEqual(values, [1, 10, 20, 30, 42, 1, 10, 20, 30, 42])
+
+    def test_QueryFactory(self):
+
+        cfg = config.Config.from_yaml([io.StringIO(_CFG1)])
+
+        for qclass in cfg.classes():
+            queries = cfg.queries(qclass)
+            for qid, factory in queries.items():
+                for i in range(10):
+                    query = factory.query()
+                    self.assertIsInstance(query, str)
+
+
+class TestMockDb(unittest.TestCase):
+
+    def test_mock_db_fetchall(self):
+
+        # can take any args
+        conn = mock_db.connect(host="host", port=4040, user='qsmaster',
+                               passwd='', db='LSST')
+
+        cursor = conn.cursor()
+
+        query = "SELECT * from Object"
+        cursor.execute(query)
+        rows = cursor.fetchall()
+        self.assertEqual(len(rows), 2)
+
+        query = "SELECT * from Object where O = 5"
+        cursor.execute(query)
+        rows = cursor.fetchall()
+        self.assertEqual(len(rows), 5)
+
+        query = "SELECT * from Ob22ject"
+        cursor.execute(query)
+        rows = cursor.fetchall()
+        self.assertEqual(len(rows), 22)
+
+    def test_mock_db_fetchmany(self):
+
+        # can take any args
+        conn = mock_db.connect(host="host", port=4040, user='qsmaster',
+                               passwd='', db='LSST')
+
+        cursor = conn.cursor()
+
+        query = "SELECT * from Object"
+        cursor.execute(query)
+        rows = cursor.fetchmany()
+        self.assertEqual(len(rows), 1)
+        rows = cursor.fetchmany()
+        self.assertEqual(len(rows), 1)
+        rows = cursor.fetchmany()
+        self.assertEqual(len(rows), 0)
+
+        query = "SELECT * from Object where O = 5"
+        cursor.execute(query)
+        rows = cursor.fetchmany(4)
+        self.assertEqual(len(rows), 4)
+        rows = cursor.fetchmany(4)
+        self.assertEqual(len(rows), 1)
+        rows = cursor.fetchmany(4)
+        self.assertEqual(len(rows), 0)
+
+        query = "SELECT * from Ob22ject"
+        cursor.execute(query)
+        rows = cursor.fetchmany(1000)
+        self.assertEqual(len(rows), 22)
+        rows = cursor.fetchmany(1000)
+        self.assertEqual(len(rows), 0)
+
+
+class TestQueryRunner(unittest.TestCase):
+
+    def test_query_runner_count(self):
+
```

Comment:
could use a docstring: what is getting tested here? I can't see any verification.

Reply:
same for the other test function in this class and the test functions in TestRunnerManager.

Reply:
👍  This test is for that it does not run forever and eventually stops. I'll add a comment.